### PR TITLE
[Feature Store]: Fix split flow

### PR DESCRIPTION
**Pull request by urihoenig**

### CODE REVIEW

File: mlrun/datastore/targets.py [hunk 1]
```diff
@@ -106,6 +106,19 @@ def add_target_states(graph, resource, targets, to_df=False, final_state=None):
...
 
 
+def _get_column_list(features, timestamp_key, key_columns):
+    column_list = None
+    if features:
+        column_list = list(features.keys())
+        if timestamp_key and timestamp_key not in column_list:
+            column_list = [timestamp_key] + column_list
+        if key_columns:
+            for key in reversed(key_columns):
```
Comment (yaronha) [inline mlrun/datastore/targets.py:116]: why r we using `reversed`? also do we need to add the index key in all targets?

File: mlrun/feature_store/api.py [hunk 2]
```diff
@@ -292,6 +292,13 @@ def infer(
 
     namespace = namespace or get_caller_globals()
     if featureset.spec.require_processing():
+        _, default_final_state, _ = featureset.graph.check_and_process_graph(
+            allow_empty=True
+        )
+        if not default_final_state:
+            raise mlrun.errors.MLRunPreconditionFailedError(
```
Comment (yaronha) [inline mlrun/feature_store/api.py:299]: i assume we still need to force the user to have default final state?

Comment (urihoenig) [reply mlrun/feature_store/api.py:299]: Forcing in addition to this exception?

File: mlrun/datastore/targets.py (hunk 1 above)
Comment (urihoenig) [reply mlrun/datastore/targets.py:116]: Both are existing logic I just wanted to preserve in my refactoring.
reversed isn't used in Parquet & the timestamp key isn't used in TSDB.

**Pull request closed without merging**